e2e/optimism_test.go

```diff
@@ -130,109 +131,139 @@ func (a *L2AltDA) ActL1Finalized(t actions.Testing) {
 	a.sequencer.ActL1FinalizedSignal(t)
 }
 
-func TestOptimismKeccak256Commitment(gt *testing.T) {
-	if !runIntegrationTests && !runTestnetIntegrationTests && !runIntegrationTestsV2 {
-		gt.Skip("Skipping test as INTEGRATION or TESTNET env var not set")
+func TestOptimismKeccak256Commitment(t *testing.T) {
+	testMatrix := testmatrix.NewTestMatrix()
+	testMatrix.AddDimension(testmatrix.NewDimension(e2e.V2Enabled, []any{true, false}))
+	testMatrix.AddDimension(testmatrix.NewDimension(e2e.Environment, []any{e2e.Local, e2e.Testnet}))
+
+	testConfigurations := testMatrix.GenerateTestConfigurations()
+	for _, testConfiguration := range testConfigurations {
+		t.Run(
+			testConfiguration.ToString(), func(t *testing.T) {
+				v2Enabled, ok := testConfiguration.GetValue(e2e.V2Enabled).(bool)
+				require.True(t, ok)
+
+				environment, ok := testConfiguration.GetValue(e2e.Environment).(e2e.TestEnvironment)
+				require.True(t, ok)
+
+				testCfg := e2e.NewTestConfig(e2e.UseMemstore(environment), v2Enabled)
+				testCfg.UseKeccak256ModeS3 = true
```

Comment:
Def like the look of this more, but unsure about having a separate matrix testConfigurations, and then another layer testCfg on top.

I guess that's the price to pay for having your untyped dynamic matrix framework. Are we sure that we need its flexibility? Would personally prefer if we started small with a typed hardcoded matrix. But maybe you see otherwise. Just feels like a lot of boilerplate for every test we need to setup...

Perhaps we can move all of this code into yet another testHarness structure? I feel like every test in this file has the same boilerplate to setup the same testmatrix and from it instantiate the testCfg right?

Reply:
In [3996e16](https://github.com/Layr-Labs/eigenda-proxy/pull/337/commits/3996e16b7f73fd0d4296f8c80e6dbd4c2d35f0a2), I did some reorganization, and added helper methods to decrease the amount of boilerplate per test. I may make some more modifications in following commits to make this more palatable, still working through your other comments.